cmake/embed_llvm.cmake

```diff
@@ -0,0 +1,164 @@
+if(EMBED_LLVM)
+  include(ExternalProject)
+  include(embed_helpers)
+
+  # TO DO
+  # Set up cross-compilation
+  # https://cmake.org/cmake/help/v3.6/manual/cmake-toolchains.7.html#cross-compiling-using-clang
+  get_host_triple(CHOST)
+  get_target_triple(CBUILD)
+
+  set(EMBED_CLANG OFF CACHE BOOL "Build Clang static libs as an ExternalProject and link to these instead of system libs.")
+  set(EMBED_LIBCLANG_ONLY OFF CACHE BOOL "Build only libclang.a, and link to system libraries statically")
+
+  if(NOT EMBED_CLANG)
+    message(AUTHOR_WARNING "Building embedded LLVM that won't be used by any target, EMBED_CLANG is not set.")
```

Comment:
So clang depends on llvm. But we also directly depend on llvm for codegen right? So I think `EMBED_CLANG=OFF && EMBED_LLVM=ON` still does useful stuff.

Reply:
If you have system clang libraries installed, you probably don't need to embed LLVM as you'll already have it installed. This will replace the system LLVM targets with the from-source version, which I figured there was no real use case for.

I can remove this warning if you think it's useful, technically it should work, but I figured it would only ever be done with EMBED_CLANG also.

Reply:
Ah yeah that makes sense. Can you leave a note (in the message or as a comment) about how clang installed implies llvm installed?

Reply:
Yeah i'll put this in the docs and reference the docs inline, I agree with your earlier comment about consolidating the documentation for this.

Reply:
I think this sholud be covered by the updated docs.